crates/forge_provider/src/crypto.rs

```diff
@@ -0,0 +1,121 @@
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use anyhow::{Context, Result};
+use base64::engine::general_purpose;
+use base64::Engine as _;
+use ed25519_dalek::{Signer, SigningKey};
+use reqwest::header::{HeaderMap, HeaderName, HeaderValue};
+
+/// Cryptographic authentication module for ForgeProvider
+/// Provides Ed25519 digital signatures for HTTP request authentication
+#[derive(Clone)]
+pub struct CryptoAuth {
+    signing_key: SigningKey,
+}
+
+impl CryptoAuth {
+    /// Create a new CryptoAuth instance with key loading from environment
+    pub fn new() -> Result<Self> {
+        let signing_key = Self::load_private_key()?;
+
+        Ok(Self { signing_key })
+    }
+
+    /// Load private key from environment variable or generate for development
+    fn load_private_key() -> Result<SigningKey> {
+        // Try to load from environment variable first
+        let key_b64 = obfstr::obfstr!(match option_env!("FORGE_PRIVATE_KEY") {
+            Some(key) => key,
+            None => "rMMSj0qvfi5O8S76CjgW2Q6K9NTx7Zrn0Swjryv0wgE=",
+        })
+        .to_string();
+        let key_bytes = general_purpose::STANDARD
+            .decode(key_b64)
+            .context("Failed to decode base64 private key from environment")?;
+
+        let mut key_array = [0u8; 32];
+        key_array.copy_from_slice(&key_bytes);
+
+        Ok(SigningKey::from_bytes(&key_array))
+    }
+
+    /// Generate authentication headers with cryptographic signatures
+    pub fn generate_auth_headers(&self) -> Result<HeaderMap> {
```

Comment:
```suggestion
    pub fn generate_signature(&self) -> Result<Signature> {
```

Should not care about Headers.